pkg/databrickssql/connector.go

```diff
@@ -0,0 +1,169 @@
+	storageURL := fmt.Sprintf("%s://%s%s", storageURI.Scheme, storageURI.Host, storageURI.Path)
+
+	return &DatabricksConnector{
+		db:  databricksDB,
+		ctx: context.Background(),
+
+		temporaryCredentials: credentials.NewCredentials(NewTemporaryCredentialsProvider(srcCredentials, awsRegion)),
+
+		storageURL: storageURL,
+		columns:    nil,
+	}, nil
+}
+
+func (dc *DatabricksConnector) InitSchema(columns []cloudstorage.TableCol) error {
+	if len(dc.columns) != 0 {
+		return nil
+	}
+	if len(columns) == 0 {
+		return errors.New("Columns in schema is empty")
+	}
+	dc.columns = columns
+	log.Info("table columns initialized", zap.Any("Columns", columns))
+	return nil
+}
+
+func (dc *DatabricksConnector) CopyTableSchema(sourceDatabase string, sourceTable string, sourceTiDBConn *sql.DB) error {
+	dropTableSQL := GenDropTableSQL(sourceTable)
+	_, err := dc.db.Exec(dropTableSQL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = dc.setColumns(sourceDatabase, sourceTable, sourceTiDBConn)
+	createTableSQL, err := GenCreateTableSQL(sourceTable, dc.columns)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	log.Info("Creating table in Databricks Warehouse", zap.String("query", createTableSQL))
+
+	_, err = dc.db.Exec(createTableSQL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	log.Info("Successfully copying table scheme", zap.String("database", sourceDatabase), zap.String("table", sourceTable))
+	return nil
+}
+
+func (dc *DatabricksConnector) LoadSnapshot(targetTable, filePrefix string, onSnapshotLoadProgress func(loadedRows int64)) error {
+	if err := LoadCSVFromS3(dc.db, dc.columns, targetTable, dc.storageURL, filePrefix, dc.temporaryCredentials); err != nil {
+		return errors.Trace(err)
+	}
+	log.Info("Successfully load snapshot", zap.String("table", targetTable), zap.String("filePrefix", filePrefix))
+	return nil
+}
+
+func (dc *DatabricksConnector) ExecDDL(tableDef cloudstorage.TableDefinition) error {
+	if len(dc.columns) == 0 {
+		return errors.New("Columns not initialized. Maybe you execute a DDL before all DMLs, which is not supported now.")
+	}
+	ddls, err := GenDDLViaColumnsDiff(dc.columns, tableDef)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(ddls) == 0 {
+		log.Info("No need to execute this DDL in Databricks", zap.String("ddl", tableDef.Query))
+		return nil
+	}
+	// One DDL may be rewritten to multiple DDLs
+	for _, ddl := range ddls {
+		_, err := dc.db.Exec(ddl)
+		if err != nil {
+			log.Error("Failed to executed DDL", zap.String("received", tableDef.Query), zap.String("rewritten", strings.Join(ddls, "\n")))
+			return errors.Annotate(err, fmt.Sprint("failed to execute", ddl))
+		}
+	}
+	// update columns
+	dc.columns = tableDef.Columns
+	log.Info("Successfully executed DDL", zap.String("received", tableDef.Query), zap.String("rewritten", strings.Join(ddls, "\n")))
+	return nil
+}
+
+func (dc *DatabricksConnector) LoadIncrement(tableDef cloudstorage.TableDefinition, uri *url.URL, filePath string) error {
```

Comment:
https://docs.databricks.com/en/sql/language-manual/sql-ref-external-tables.html 
can we use external table?

Reply:
I used AWS Security Token Service, a temporary access key, secret key, and token. If we use an external table, the user needs to offer a `CREDENTIAL NAME` to tidb2dw. Is it cool? I don't know which one is better.

Reply:
IMO, performance is most important norm. With external table, we can reduce one copy each time load a cdc file? /cc @breezewish

Reply:
And in cloud, we also require long term credentials to create CDC changefeed.

Reply:
It can reduce one copy. If we require long-term credentials whatever the scenes are, it will make sense to add a credential name for the user.

Reply:
There is another thing I found but didn't reveal in document of databricks. They don't support the `BINARY` type in the external table with the CSV file. Should I add this thing to our document and drop the support for the `BINARY`?
![image](https://github.com/pingcap-inc/tidb2dw/assets/26212551/c79b41aa-ba31-4d78-948b-c236647a88fc)

Reply:
@Lloyd-Pottiger I already supported the external table and specified the credential name in this commit https://github.com/pingcap-inc/tidb2dw/pull/66/commits/6daa6e86a57f11c91f54670bae803e7f8865abba. If you have any suggestions, please let me know.

Reply:
> There is another thing I found but didn't reveal in document of databricks. They don't support the `BINARY` type in the external table with the CSV file. Should I add this thing to our document and drop the support for the `BINARY`?
> ![image](https://user-images.githubusercontent.com/26212551/270670191-c79b41aa-ba31-4d78-948b-c236647a88fc.png)

yes, add into `docs/databricks.md`

Reply:
Got it. Added it to `docs/databricks.md` and dropped the `binary` and `varbinary` support in `pkg/databrickssql/types.go` map.